### ISSUE TITLE
Dataset shape

### ISSUE THREAD

**Issue by RhianTaylor6** [Dref360/spectral_metric#4]
Hello,

Trying to use this to figure out the CSG of MNIST Fashion loaded from keras. No matter what pre-processing I try I keep getting a shape error. I figure it should be similar to whatever pre-processing was used on normal MNIST and was hoping you could help?

Here is the error I am getting:
boolean index did not match indexed array along dimension 1; dimension is 28 but corresponding boolean dimension is 10

**Comment by Dref360** [comment]
Hello,

Do you have some code I could look at?


We expect an array with shape [N, num_features] so you need to flatten the images.

**Comment by RhianTaylor6** [comment]
Hi this was my latest attempt, but I don't think I am flattening the images here...not entirely sure how haha

#from spectral_metric.estimator import CumulativeGradientEstimator
#from spectral_metric.visualize import make_graph
from keras.datasets import fashion_mnist,mnist
from keras.utils import to_categorical
import numpy as np


#(X_train, y_train), test_data = fashion_mnist.load_data()
(X_train, y_train), test_data = fashion_mnist.load_data()

# Normalize the images.
X_train = (X_train / 255) - 0.5
# reshape dataset to have a single channel
X_train = X_train.reshape((X.shape[0],28,28,1))
y_train = to_categorical(y_train)


estimator = CumulativeGradientEstimator()
estimator.fit(data=X_train, target=y_train)
csg = estimator.csg  # The actual complexity values.
estimator.evals, estimator.evecs  # The eigenvalues and vectors.

**Comment by Dref360** [comment]
You can flatten the images with:

```python
X_train = X_train.reshape((X.shape[0],-1))
```

**Comment by RhianTaylor6** [comment]
then I get a similar error: boolean index did not match indexed array along dimension 1; dimension is 784 but corresponding boolean dimension is 10

**Comment by Dref360** [comment]
Ah I see the issue, you must not call `to_categorical` here.
And we expect an array with a single dimension.

```python
y_train = y_train.reshape([-1])
```

**Comment by RhianTaylor6** [comment]
Thank you so much! That worked and MNIST Fashion has a CSG of 0.61860741 in case anyone else needs to know!

**Comment by Dref360** [comment]
Awesome! So a bit easier than notMNIST

![image](https://user-images.githubusercontent.com/8976546/150164283-ad795a15-de34-4d2f-b80a-c5594f0b5d85.png)

**Comment by RhianTaylor6** [comment]
Yep! And that kinda makes sense doesn't it! :)

**Comment by RhianTaylor6** [comment]
Hello again, just had a quick question about the metrics in your paper, did you take an average for CSGs at all over a certain number of calculations?

**Comment by Dref360** [comment]
This is the average over 20 runs I think? But the standard deviation was very small as you can see in Figure 2.

**Comment by RhianTaylor6** [comment]
That's amazing thank you so much! As you may have guessed I am using the CSG from your paper in my own work and I just want to make sure the MNIST Fashion CSG value is inline with the others1 I really appreciate your help!

**Comment by Dref360** [comment]
WHen it is available, send me a link and I'll add it to the README :)

**Comment by RhianTaylor6** [comment]
I will thank you!

**Comment by RhianTaylor6** [comment]
Hello again,
I have been trying to calculate the CSG of CIFAR-10 because I want the range to display on a chart. In your paper it is reported as being 1 but I am getting it as over 3 and I am not sure why, would you be so kind as to assist me again?

**Comment by Dref360** [comment]
Yeah sure.

For the paper, we got CIFAR10 embeddings using an autoencoder and ran t-SNE on it. We used MultiCoreTSNE.

CNN encoder code: https://github.com/Dref360/spectral_metric/blob/master/experiments/embedding/cnn_autoencoder.py
t-SNE code: https://github.com/Dref360/spectral_metric/blob/master/experiments/embedding/tsne.py

To compare datasets, they need to be from similar embedding and in the paper, we only showed scores for CNN+t-SNE I think.
